tastypie/resources.py

```diff
@@ -254,6 +254,11 @@ def _handle_500(self, request, exception):
         data = {
             "error_message": getattr(settings, 'TASTYPIE_CANNED_ERROR', "Sorry, this request could not be processed. Please try again later."),
         }
+
+        # If the exception comes with an error, show that instead of "Sorry, this request..."
+        if getattr(exception, 'message', None):
```

Comment:
Maybe only do this for exceptions that also have the attribute "safe_for_public_viewing" or similar?  That way you can throw custom exceptions with that attribute set and expect to see them with DEBUG=False, but others are masked by default.